### ISSUE TITLE
[Bug] QueryService Service Account uses the same annotations for ServiceAccount as for Cost Analyzer

### ISSUE THREAD

**Issue by rajatvig** [kubecost/cost-analyzer-helm-chart#2784]
### Kubecost Helm Chart Version

1.106.4

### Kubernetes Version

1.27

### Kubernetes Platform

GKE

### Description

QueryService Service Account uses the same annotations for ServiceAccount as for Cost Analyzer. This creates a problem with Workload Identity as we need different annotations for both service accounts.

### Steps to reproduce

1. Add annotations for ServiceAcount in helm values.
2. Is copied for Query Service SA

### Expected behavior

QueryService should have its own SA annotations

### Impact

This creates a problem with Workload Identity as we need different annotations for both service accounts.

### Troubleshooting

- [X] I have read and followed the [issue guidelines](https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/ISSUE_GUIDELINES.md) and this is a bug impacting only the Helm chart.
- [X] I have searched other issues in this repository and mine is not recorded.

**Comment by chipzoller** [comment]
Can you please show a values snippet where you added these annotations?

**Comment by rajatvig** [comment]
It is simple as

```yaml
serviceAccount:
  create: true 
  annotations:
    name: kc-test
```

If you look at the sources in https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/templates/query-service-service-account-template.yaml#L11-L14 it uses the chart values for the default SA and nto for query service.

**Comment by chipzoller** [comment]
This was fixed in #2766
